src/afrolabs/components/kafka.clj

```diff
@@ -1419,10 +1494,22 @@
            ktable-segment-ms
            ktable-min-cleanable-dirty-ratio
            ktable-max-compaction-lag-ms
-           recreate-topics-with-bad-config]
-    :or   {recreate-topics-with-bad-config false}}]
-
-  (let [nr-of-partitions (or (when (and nr-of-partitions
+           update-topics-with-bad-config
+           ktable-compaction-policy]
+    :or   {ktable-compaction-policy        "compact"}}]
+
+  (when-not (nil? (:recreate-topics-with-bad-config cfg))
+    (log/info "Ktable asserter option `recreate-topics-with-bad-config` has been deprecated in favour of `update-topics-with-bad-config`."))
+
+  (let [update-topics-with-bad-config (or (when-not (nil? update-topics-with-bad-config)
+                                            update-topics-with-bad-config)
+                                          (let [recreate-topics-with-bad-config (:recreate-topics-with-bad-config cfg)]
+                                            (when-not (nil? recreate-topics-with-bad-config)
+                                              recreate-topics-with-bad-config)))
```

Comment:
Could these not just be a plain `(or update-topics-with-bad-config (:recreate-topics-with-bad-config cfg))`

This is also defined very far from where it is used, and it may make it easier to read if it was just inlined where it is used, or if this needs all the `when-let`s then at least moved to just above the `if` and changed to `update-topics-with-bad-config?`

Reply:
`update-topics-with-bad-config` is a parameter whose binding is being shadowed, so I want to make sure the rebinding is done close to the function definition because semantically it's still a parameter.

It's (final) value depends on the presence of the deprecated old field and I want to make that obvious (to the maintainer).

The value of the parameter may be set as `false`, in which case that is the value I actually want to use. The simpler `(or update-topics-with-bad-config ...)` conflates `nil` (unset) with `false` (set). It will ignore an intentionally set `false` value, which is not the behaviour I want. If I set this parameter value to `false` then _that_ is what I want to happen and the deprecated `recreate...` parameter must be ignored entirely.

I have some other instantiations of this class which still uses the old value, but I want to transparently upgrade those invocations to update-in-place by using `recreate-topics-with-bad-config` values if they are set to `true`.